src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java

```diff
@@ -198,6 +198,9 @@ private final void closeStatements()
             }
             catch (SQLException e) {
                checkException(e);
+               if (delegate == ClosedConnection.CLOSED_CONNECTION) {
+                  break; //connection closed in checkException
```

Comment:
Actually, it is infrequent enough for there to be unclosed Statements, especially with the prevalence of persistence frameworks (Hibernate, JPA, Play/Slick, etc.) that I'd be just as happy adding ``delegate != ClosedConnection.CLOSED_CONNECTION`` to the *for loop* conditional.